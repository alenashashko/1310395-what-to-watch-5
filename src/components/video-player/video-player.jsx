import React, {PureComponent, createRef} from 'react';

import proptypes from '../../type';

class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);

    this._timerId = null;
    this._videoRef = createRef();

    this.state = {
      isDelayCompleted: false,
      isVideoReadyToPlay: false
    };
  }

  componentDidMount() {
    const video = this._videoRef.current;

    video.src = this.props.videoSrc;

    video.oncanplaythrough = () => {
      this.setState({
        isVideoReadyToPlay: true
      });
    };

    this._timerId = setTimeout(() => this.setState({
      isDelayCompleted: true
    }), 1000);
  }

  componentDidUpdate() {
    const video = this._videoRef.current;

    if (this.state.isDelayCompleted && this.state.isVideoReadyToPlay) {
      video.play();
    }
  }

  componentWillUnmount() {
    const video = this._videoRef.current;

    video.oncanplaythrough = null;

    clearTimeout(this._timerId);
  }

  render() {
    return (
      <video width="280" height="175"
        ref={this._videoRef}
        muted
        poster={this.props.pictureSrc}>
      </video>
    );
  }
}

VideoPlayer.propTypes = {
  videoSrc: proptypes.src,
  pictureSrc: proptypes.src
};

export default VideoPlayer;
