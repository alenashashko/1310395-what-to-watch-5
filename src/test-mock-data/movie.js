export const movie = {
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
};

export const movies = [{
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
},
{
  id: `2`,
  title: `Revenant`,
  picture: `/img/revenant.jpg`,
  backgroundPicture: `/img/revenant.jpg`,
  backgroundColor: `#bdad8f`,
  poster: `/img/revenant.jpg`,
  genre: `Romance`,
  year: 2010,
  ratingScore: 5.9,
  ratingCount: 850,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 145,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  director: `Robert Zemeckis`,
  starring: [`Will Smith`, `Michael J. Fox`, `Harrison Ford`,
    `Sandra Bullock`, `Betty White`, `Tom Hanks`, `Denzel Washington`],
  isFavorite: false
}
];

export const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`,
  `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];

export const genre = `All genres`;
