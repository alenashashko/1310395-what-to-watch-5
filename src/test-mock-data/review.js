import {adaptCommentToClient} from '../services/adapters';

export const apiReviews = [
  {
    id: 1,
    user: {
      id: 11,
      name: `Jack`
    },
    rating: 2.4,
    comment: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    date: `2020-11-03T13:38:44.769Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      name: `Isaac`
    },
    rating: 7.2,
    comment: `A movie that will take you to another world full of emotions.`,
    date: `2020-10-23T13:38:44.769Z`
  }
];

export const review = adaptCommentToClient(apiReviews[0]);

export const reviews = apiReviews.map(adaptCommentToClient);

