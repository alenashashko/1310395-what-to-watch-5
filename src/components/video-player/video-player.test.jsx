import React from 'react';
import renderer from 'react-test-renderer';

import {VideoPlayer} from './video-player';
import {movie} from '../../test-mock-data/movie';

const noop = () => {};

it(`Should VideoPlayer render correctly`, () => {
  const tree = renderer
    .create(
        <VideoPlayer
          videoRef={React.createRef()}
          pictureSrc={movie.picture}
          onCanPlayThrough={noop}
        />).toJSON();

  expect(tree).toMatchSnapshot();
});
