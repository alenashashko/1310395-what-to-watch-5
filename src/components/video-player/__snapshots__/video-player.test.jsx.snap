// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should VideoPlayer render correctly 1`] = `
<video
  height="175"
  muted={true}
  onCanPlayThrough={[Function]}
  poster="/img/orlando.jpg"
  width="280"
/>
`;
